File: rb/features.rb
# Disney SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module DisneyFeatures
  def self.make_feature(name)
    case name
    when "base"
      DisneyBaseFeature.new
    when "test"
      DisneyTestFeature.new
    else
      DisneyBaseFeature.new
    end
  end
end

File: py/features.py
# Disney SDK feature factory

from feature.base_feature import DisneyBaseFeature
from feature.test_feature import DisneyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DisneyBaseFeature(),
        "test": lambda: DisneyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
